### ISSUE TITLE
Correct approach to intercepting CTRL-C

### ISSUE THREAD

**Issue by Ezekiel-DA** [SBoudrias/Inquirer.js#1030]
First: thank you for this wonderful library. I've been using it for a while for some internal tools at work and it's made creating CLI tools for my coworkers so much easier, I tend to reach for it out of habit for any CLI projects now!

Now, a question / issue: I'm attempting to do some processing in a CLI Node app on SIGINT / Ctrl-C, mostly on Windows. I've tested this with both Inquirer 7.3.3 and 8.1.2.

It appears that Inquirer captures and "buries" SIGINT (at least I'm assuming this is what's happening, in `onForceClose()` in `packages/inquirer/lib/ui/baseUI.js`.

I see an existing issue with a suggestion of attaching an event handler to stdin and passing that in to `inquirer.prompt()` [here](https://github.com/SBoudrias/Inquirer.js/issues/405#issuecomment-268076368), but as far as I can tell `prompt()` doesn't / no longer takes that input.

Based on @SBoudrias' suggestion in that same issue, I also attempted to capture SIGINT myself via `readline()`, in a way that Stack Overflow suggests is appropriate for Windows:

```
  if (process.platform === 'win32') {
    require('readline')
      .createInterface({
        input: process.stdin,
        output: process.stdout
      })
      .on('SIGINT', function () {
          // my processing here
      })
  }
```

This actually works, __but__: all Inquirer prompts are now duplicating their output after they receive input, if that makes sense, e.g.:
```
? Password? [hidden]
? Password? **************
```
Without this hack to intercept CTRL-C, the output used to simply be:
```
? Password? **************
```
instead of this new duplication of the question and response. This happens for all prompt types btw, this isn't specific to passwords, it was just an easy example.

Any suggestions on what I'm doing wrong? or any possibility on getting customizable exit behavior from the library more cleanly?

Thanks!

**Comment by SBoudrias** [comment]
I'm not too sure, but here's a suggestion;

One of the major issue I've had is always clearing prompts because it's easy to loose the line to clear. I wonder if somehow your change would print an extra line `\n` in the console. This would mean that when inquirer clears the previous line to render the new content, it clears the empty line instead of the previous line. I know `ctrl-C` usually adds an extra empty line, so this could come from there.

**Comment by Ezekiel-DA** [comment]
Thanks for the reply!

That was sort of my guess too, but since I was unfamiliar with the inner workings of readline or inquirer, I went with the approach above (`readline.createInterface()`). I did notice that not attaching stdout to it (removing `output: process.stdout` from the snippet above) prevented the double prompt issue, but it also prevented CTRL-C from being caught. I'm not entirely sure why, I don't see how attaching a writable is necessary in this case.

Anyway: inspired by your comment, I figured my way of dealing with `readline` was the problem; that led me to realize that the snippet offered in the comment I linked above does what I need, even if I can't pass the modified stdin to Inquirer as suggested there:

```
let stdin = process.stdin;
stdin.on("data", (key) => {
    if (key == "\u0003") {
        LOG("catching ctrl+c");
    }
});

// inquirer.prompt(questions, {input: stdin}); // <== this doesn't work, but isn't actually necessary anyway?
```
Doing this gives me a way to handle CTRL-C and does not appear to be breaking Inquirer in any way that I can see!

Thank you for the help / rubber ducking!
